### ISSUE TITLE
Three enhancements: @include, @charset, and css file extension

### ISSUE THREAD

**Issue by GoogleCodeExporter** [chrsmithdemos/minify#109]
```
Minify version: 2.1.2
PHP version: 5.1.6

What steps will reproduce the problem?
1. Aggregate CSS-files that includes @charset and @include statements
2. Link the the result via: <meta http-equiv="Content-Style-Type" 
content="text/css" /><link 
href="/public_tools/min/g=myCSS" rel="stylesheet" type="text/css" />
3. Open the page with Safari 4 under OS X 10.5.6

Expected output: Nicely rendered web page.

Actual output:
 * Invalid CSS file since @charset is only allowed at the beginning
 * No CSS layout since Safari is confused how to handle "/min/g=myCSS"

Did any unit tests FAIL? (Please do not post the full list)
 * No tested

Please provide any additional information below. If this has to do with URI 
rewriting, please 
include your DOCUMENT_ROOT.

I propose the following changes:

1.) RewriteRule ^([a-z])_(.*)\.css$  index.php?$1=$2  [L,NE]
=> this will treat the script output a a regular CSS file (compression, content 
type, ...) and Safari 
is happy with it

2.) At the end of _process($css): $css = preg_replace('/@charset ".*";/', '', 
$css);
=> this will remove the @charset statements (maybe you could add one at the 
beginning of the 
output based on the input encoding)

3.) At the end of _process($css): $css = preg_replace('/@import url\((.*)\);/', 
'/*WARNING: 
IMPORT of "\1!"*/', $css);
=> Just as a very simple default solution for this common problem
```

---

Original issue reported on code.google.com by `Alexande...@googlemail.com` on 29 Apr 2009 at 2:45

**Comment by GoogleCodeExporter** [comment]
```
1. Minify should be sending Content-Type: text/
css;charset=UTF-8. As long as this header is sent, the browser shouldn't care 
about 
the URL. On your page w/o CSS showing, what's the Content-Type sent? (You'll 
need an 
HTTP sniffer).

2. Re: @charset, I want to keep Minify simple: Use UTF-8 in your files. You can 
change the charset sent by setting $min_serveOptions['contentTypeCharset'] in 
config.php, but the minifiers can't handle encodings like UTF-16 anyway. I'm 
willing 
to discuss this more on the Google group.

3. Something like this already implemented in 2.1.2:
http://code.google.com/p/minify/source/browse/tags/release_2.1.2/min/config.php#
69
Minify should prepend a note to the CSS if you've ended up with an @import not 
at 
the top.
```

---

Original comment by `mrclay....@gmail.com` on 29 Apr 2009 at 8:24
- Changed title: **Three enhancements: @include, @charset, and css file extension**
- Changed state: **NeedInfo**

**Comment by GoogleCodeExporter** [comment]
```
@include essentially dup of Issue 50.
```

---

Original comment by `mrclay....@gmail.com` on 1 Jul 2009 at 8:29
- Changed state: **Invalid**
